AEPMessaging/Tests/UnitTests/BatchedPropositionInteractionTests.swift

```diff
@@ -0,0 +1,225 @@
+    var proposition1: MockProposition!
+    var proposition2: MockProposition!
+    var propositionItem1: PropositionItem!
+    var propositionItem2: PropositionItem!
+
+    
+    override func setUp() {
+        super.setUp()
+        
+        // Setup common test data
+        scopeDetails1 = ["key1": "value1"]
+        scopeDetails2 = ["key2": "value2"]
+        propositionItem1 = PropositionItem(itemId: "item1", schema: .contentCard, itemData: [:])
+        propositionItem2 = PropositionItem(itemId: "item2", schema: .contentCard, itemData: [:])        
+        proposition1 = MockProposition(uniqueId: "proposition1", scope: "scope1", scopeDetails: scopeDetails1, items: [propositionItem1])
+        proposition2 = MockProposition(uniqueId: "proposition2", scope: "scope2", scopeDetails: scopeDetails2, items: [propositionItem2])
+
+        propositionItem1.proposition = proposition1
+        propositionItem2.proposition = proposition2
+        
+    }
+    
+    func testBatchPropositionInteractionXDM_forDisplayEvent() {
+        // setup
+        let propositionItems = [propositionItem1!, propositionItem2!]
+        
+        // test
+        let batchedInteraction = BatchedPropositionInteraction(
+            eventType: .display,
+            interaction: nil,
+            propositionItems: propositionItems
+        )
+
+        let xdm = batchedInteraction.generateXDM()
+        
+        // Then
+        let expectedXdm: [String: Any] = [
+            "eventType": "decisioning.propositionDisplay",
+            "_experience": [
+                "decisioning": [
+                    "propositions": [
+                        [
+                            "id": "proposition1",
+                            "scope": "scope1",
+                            "scopeDetails": [
+                                "key1": "value1"
+                            ],
+                            "items": [
+                                [
+                                    "id": "item1"
+                                ]
+                            ]
+                        ],
+                        [
+                            "id": "proposition2",
+                            "scope": "scope2",
+                            "scopeDetails": [
+                                "key2": "value2"
+                            ],
+                            "items": [
+                                [
+                                    "id": "item2"
+                                ]
+                            ]
+                        ]
+                    ],
+                    "propositionEventType": [
+                        "display" : 1
+                    ]
+                ]
+            ]
+        ]
+        
+        // Using NSDictionary comparison as seen in other tests
+        XCTAssertTrue(NSDictionary(dictionary: xdm).isEqual(to: expectedXdm), 
+                      "XDM should match expected structure")
```

Comment:
we should use the assertions available in `AEPTestUtils`.  see `MessagingFunctionalTests` for examples.

Reply:
yup done.